// muix/common/root.js
System.register(["tslib", "react", "./icons/index", "./styles/typography", "./primitives/text-view", "./primitives/icon", "./components/button", "./styles/component-x", "./components/button-demo", "muix-primitives", "muix-styles", "muix-components/Button/Button"], function (exports_1, context_1) {
    "use strict";
    var __moduleName = context_1 && context_1.id;
    var tslib_1, react_1, index_1, typography_1, text_view_1, icon_1, button_1, component_x_1, button_demo_1, muix_primitives_1, muix_styles_1, Button_1, apps, theme, AppRoot, AppItem;
    return {
        setters: [
            function (tslib_1_1) {
                tslib_1 = tslib_1_1;
            },
            function (react_1_1) {
                react_1 = react_1_1;
            },
            function (index_1_1) {
                index_1 = index_1_1;
            },
            function (typography_1_1) {
                typography_1 = typography_1_1;
            },
            function (text_view_1_1) {
                text_view_1 = text_view_1_1;
            },
            function (icon_1_1) {
                icon_1 = icon_1_1;
            },
            function (button_1_1) {
                button_1 = button_1_1;
            },
            function (component_x_1_1) {
                component_x_1 = component_x_1_1;
            },
            function (button_demo_1_1) {
                button_demo_1 = button_demo_1_1;
            },
            function (muix_primitives_1_1) {
                muix_primitives_1 = muix_primitives_1_1;
            },
            function (muix_styles_1_1) {
                muix_styles_1 = muix_styles_1_1;
            },
            function (Button_1_1) {
                Button_1 = Button_1_1;
            }
        ],
        execute: function () {
            apps = [
                { title: 'Typography', app: typography_1.default },
                { title: 'ButtonDemo', app: button_demo_1.default, overridesNew: button_demo_1.overridesNew },
                { title: 'Button', app: button_1.default },
                { title: 'Icons', app: index_1.default },
                { title: 'TextView', app: text_view_1.default },
                { title: 'Icon', app: icon_1.default },
                //{ title: 'Shadows', app: Shadows },
                { title: 'ComponentX', app: component_x_1.default },
            ];
            theme = muix_styles_1.createMuiTheme({
                palette: {
                    type: 'dark',
                }
            });
            AppRoot = /** @class */ (function (_super) {
                tslib_1.__extends(AppRoot, _super);
                function AppRoot() {
                    var _this = _super !== null && _super.apply(this, arguments) || this;
                    _this.state = { appIndex: 0, rtl: false, light: false };
                    return _this;
                }
                AppRoot.prototype.render = function () {
                    var _this = this;
                    var appIndex = this.state.appIndex;
                    var App = apps[appIndex].app;
                    var root = react_1.default.createElement(muix_primitives_1.View, { style: { flex: 1, paddingTop: 24 } },
                        react_1.default.createElement(muix_styles_1.MuiThemeProvider, { theme: theme },
                            react_1.default.createElement(muix_primitives_1.View, { classes: function (theme) { return ({ root: { flexDirection: 'row', flexWrap: 'wrap', backgroundColor: theme.palette.background.appBar, flexShrink: 0 } }); } }, apps.map(function (app, idx) { return react_1.default.createElement(AppItem, { key: idx, idx: idx, active: idx === appIndex, appRoot: _this }); }))),
                        react_1.default.createElement(muix_primitives_1.Typography, { type: 'display2' }, apps[appIndex].title),
                        react_1.default.createElement(App, null));
                    var root2 = react_1.default.createElement(typography_1.default, null);
                    return react_1.default.createElement(muix_styles_1.AppContainer, { key: appIndex, themeOptions: { overridesX: apps[appIndex].overridesNew } }, root);
                };
                return AppRoot;
            }(react_1.default.Component));
            AppItem = /** @class */ (function (_super) {
                tslib_1.__extends(AppItem, _super);
                function AppItem() {
                    return _super !== null && _super.apply(this, arguments) || this;
                }
                AppItem.prototype.render = function () {
                    var _a = this.props, idx = _a.idx, active = _a.active, appRoot = _a.appRoot;
                    return react_1.default.createElement(Button_1.default, { color: 'primary', disabled: active, onClick: function () { return appRoot.setState(function (state) { return (tslib_1.__assign({}, state, { appIndex: idx })); }); } }, apps[idx].title);
                };
                return AppItem;
            }(react_1.default.PureComponent));
            exports_1("default", AppRoot);
        }
    };
});
