// muix/common/styles/typography.js
System.register(["react", "muix-styles", "material-ui/colors/orange", "material-ui/colors/green", "material-ui/colors/cyan", "muix-primitives"], function (exports_1, context_1) {
    "use strict";
    var __moduleName = context_1 && context_1.id;
    var react_1, muix_styles_1, orange_1, green_1, cyan_1, muix_primitives_1, theme, app2, app;
    return {
        setters: [
            function (react_1_1) {
                react_1 = react_1_1;
            },
            function (muix_styles_1_1) {
                muix_styles_1 = muix_styles_1_1;
            },
            function (orange_1_1) {
                orange_1 = orange_1_1;
            },
            function (green_1_1) {
                green_1 = green_1_1;
            },
            function (cyan_1_1) {
                cyan_1 = cyan_1_1;
            },
            function (muix_primitives_1_1) {
                muix_primitives_1 = muix_primitives_1_1;
            }
        ],
        execute: function () {
            theme = muix_styles_1.createMuiTheme({
                typography: function (palette) { return ({
                    caption: {
                        color: cyan_1.default.A700,
                        fontSize: 28,
                        fontWeight: '500',
                        native: {
                            fontSize: 22,
                        }
                    }
                }); },
                palette: {
                    primary: orange_1.default,
                    secondary: green_1.default
                },
                overridesX: {
                    MuiTypography: {
                        colorAccent: { color: cyan_1.default.A700, fontWeight: '500', fontSize: 18 },
                        display2: { color: cyan_1.default.A700 },
                    }
                }
            });
            app2 = function () { return react_1.default.createElement(muix_primitives_1.Typography, { classes: { style: { fontWeight: '500', textDecorationLine: 'line-through', color: 'blue', marginTop: 24 } } }, "TEXT"); };
            app = function () { return react_1.default.createElement(muix_primitives_1.ScrollView, null,
                react_1.default.createElement(muix_primitives_1.Typography, { paragraph: true, type: 'headline', style: { textDecorationLine: 'underline' } }, "noWrap"),
                react_1.default.createElement(muix_primitives_1.Typography, { noWrap: true, style: { width: 200, flexShrink: 0 } }, "noWrap noWrap noWrap noWrap noWrap noWrap noWrap noWrap noWrap noWrap noWrap noWrap noWrap noWrap "),
                react_1.default.createElement(muix_primitives_1.Typography, { paragraph: true, type: 'headline', style: { textDecorationLine: 'underline' } }, "MARGINS"),
                react_1.default.createElement(muix_primitives_1.Typography, { paragraph: true }, "paragraph paragraph paragraph paragraph paragraph paragraph paragraph paragraph paragraph paragraph paragraph paragraph paragraph paragraph paragraph paragraph "),
                react_1.default.createElement(muix_primitives_1.Typography, { paragraph: true }, "paragraph paragraph paragraph paragraph paragraph paragraph paragraph paragraph paragraph paragraph paragraph paragraph paragraph paragraph paragraph paragraph "),
                react_1.default.createElement(muix_primitives_1.Typography, { gutterBottom: true }, "gutterBottom gutterBottom gutterBottom gutterBottom gutterBottom gutterBottom gutterBottom gutterBottom gutterBottom gutterBottom gutterBottom "),
                react_1.default.createElement(muix_primitives_1.Typography, { gutterBottom: true }, "gutterBottom gutterBottom gutterBottom gutterBottom gutterBottom gutterBottom gutterBottom gutterBottom gutterBottom gutterBottom gutterBottom "),
                react_1.default.createElement(muix_primitives_1.Typography, { paragraph: true, type: 'headline', style: { textDecorationLine: 'underline' } }, "FONTS (native only)"),
                react_1.default.createElement(muix_primitives_1.Typography, { style: { fontFamily: 'Roboto' } }, "Roboto"),
                react_1.default.createElement(muix_primitives_1.Typography, { style: { fontFamily: 'Roboto_Light' } }, "Roboto_Light"),
                react_1.default.createElement(muix_primitives_1.Typography, { style: { fontFamily: 'Roboto_Medium' } }, "Roboto_Medium"),
                react_1.default.createElement(muix_primitives_1.Typography, { paragraph: true, type: 'headline', style: { textDecorationLine: 'underline' } }, "COLORS"),
                react_1.default.createElement(muix_primitives_1.Typography, null, "undefined"),
                react_1.default.createElement(muix_primitives_1.Typography, { color: 'inherit' }, "inherit"),
                react_1.default.createElement(muix_primitives_1.Typography, { color: 'accent' }, "accent"),
                react_1.default.createElement(muix_primitives_1.Typography, { color: 'error' }, "error"),
                react_1.default.createElement(muix_primitives_1.Typography, { color: 'default' }, "default"),
                react_1.default.createElement(muix_primitives_1.Typography, { color: 'secondary' }, "secondary"),
                react_1.default.createElement(muix_primitives_1.Typography, { color: 'primary' }, "primary"),
                react_1.default.createElement(muix_primitives_1.Typography, { paragraph: true, type: 'headline', style: { textDecorationLine: 'underline' } }, "TYPES"),
                react_1.default.createElement(muix_primitives_1.Typography, null, "undefined"),
                react_1.default.createElement(muix_primitives_1.Typography, { type: 'body1', align: "right" }, "body1, align=right"),
                react_1.default.createElement(muix_primitives_1.Typography, { type: 'body2', align: "center" }, "body2, align=center"),
                react_1.default.createElement(muix_primitives_1.Typography, { type: 'button', align: "left" }, "button, align=left"),
                react_1.default.createElement(muix_primitives_1.Typography, { type: 'caption' }, "caption"),
                react_1.default.createElement(muix_primitives_1.Typography, { paragraph: true, type: 'headline' }, "headline"),
                react_1.default.createElement(muix_primitives_1.Typography, { type: 'subheading' }, "subheading"),
                react_1.default.createElement(muix_primitives_1.Typography, { type: 'title' }, "title"),
                react_1.default.createElement(muix_primitives_1.Typography, { type: 'display1' }, "display1"),
                react_1.default.createElement(muix_primitives_1.Typography, { type: 'display2' }, "display2"),
                react_1.default.createElement(muix_primitives_1.Typography, { type: 'display3' }, "display3"),
                react_1.default.createElement(muix_primitives_1.Typography, { type: 'display4' }, "display4"),
                react_1.default.createElement(muix_primitives_1.Typography, { paragraph: true, type: 'headline', style: { textDecorationLine: 'underline' } }, "WITH CLASSES"),
                react_1.default.createElement(muix_primitives_1.Typography, { classes: { style: { fontWeight: '500', textDecorationLine: 'line-through', color: 'blue' } } }, "TEXT"),
                react_1.default.createElement(muix_primitives_1.Typography, { paragraph: true, type: 'headline', style: { textDecorationLine: 'underline' } }, "WITH THEME"),
                react_1.default.createElement(muix_styles_1.MuiThemeProvider, { theme: theme },
                    react_1.default.createElement(muix_primitives_1.View, null,
                        react_1.default.createElement(muix_primitives_1.Typography, { color: 'secondary' }, "secondary"),
                        react_1.default.createElement(muix_primitives_1.Typography, { color: 'primary' }, "primary"),
                        react_1.default.createElement(muix_primitives_1.Typography, { type: 'caption' }, "caption"),
                        react_1.default.createElement(muix_primitives_1.Typography, { type: 'display2' }, "display2"),
                        react_1.default.createElement(muix_primitives_1.Typography, { color: 'accent' }, "accent"),
                        react_1.default.createElement(muix_primitives_1.Typography, { color: 'accent', style: { color: 'red' } }, "accent red"),
                        react_1.default.createElement(muix_primitives_1.Typography, { color: 'primary', classes: { style: { fontWeight: '500', textDecorationLine: 'line-through' } } }, "primary, WITH CLASSES")))); };
            exports_1("default", app);
        }
    };
});
