// muix/common/components/button-demo.js
System.register(["react", "muix-components/Button/Button", "muix-styles", "muix-primitives"], function (exports_1, context_1) {
    "use strict";
    var __moduleName = context_1 && context_1.id;
    var react_1, Button_1, muix_styles_1, muix_primitives_1, theme, rootView, app2, root, overridesNew, app;
    return {
        setters: [
            function (react_1_1) {
                react_1 = react_1_1;
            },
            function (Button_1_1) {
                Button_1 = Button_1_1;
            },
            function (muix_styles_1_1) {
                muix_styles_1 = muix_styles_1_1;
            },
            function (muix_primitives_1_1) {
                muix_primitives_1 = muix_primitives_1_1;
            }
        ],
        execute: function () {
            theme = muix_styles_1.createMuiTheme({
                overridesX: function (theme) { return ({
                    MuiButton: {
                        raisedAccent: {
                            backgroundColor: 'brown',
                            $web: { '&:hover': { backgroundColor: 'brown' } }
                        }
                    }
                }); }
            });
            rootView = { flexDirection: 'row', flexWrap: 'wrap', justifyContent: 'flex-start', alignItems: 'stretch', padding: 10 };
            app2 = function (props) { return (react_1.default.createElement(muix_primitives_1.View, { classes: { root: rootView } },
                react_1.default.createElement(Button_1.default, { color: 'accent', raised: true, classes: function (theme) { return ({ raisedAccent: { backgroundColor: 'green', $web: { '&:hover': { backgroundColor: 'green', } } } }); } }, "classes"),
                react_1.default.createElement(muix_styles_1.MuiThemeProvider, { theme: theme },
                    react_1.default.createElement(Button_1.default, { color: 'accent', raised: true }, "theme")))); };
            root = { root: { margin: theme.spacing.unit } };
            //const root = {  }
            exports_1("overridesNew", overridesNew = function (theme) { return ({ MuiButton: root, MuiButtonIconLeft: root, MuiButtonIconRight: root }); });
            app = function (props) { return react_1.default.createElement(muix_primitives_1.ScrollView, null,
                react_1.default.createElement(muix_primitives_1.View, { classes: { root: rootView } },
                    react_1.default.createElement(Button_1.default, null, "Default"),
                    react_1.default.createElement(Button_1.default, { color: "primary" }, "Primary"),
                    react_1.default.createElement(Button_1.default, { color: "accent" }, "Accent"),
                    react_1.default.createElement(Button_1.default, { disabled: true }, "Disabled"),
                    react_1.default.createElement(Button_1.default, { href: "#flat-buttons" }, "Link"),
                    react_1.default.createElement(Button_1.default, { disabled: true, href: "/" }, "Link disabled"),
                    react_1.default.createElement(Button_1.default, { dense: true }, "Dense"),
                    react_1.default.createElement(Button_1.default, { onClick: function () { return alert('Here I am'); } }, "Do something")),
                react_1.default.createElement(muix_primitives_1.View, { classes: { root: rootView }, style: { backgroundColor: theme.palette.primary[500] } },
                    react_1.default.createElement(Button_1.default, { color: "contrast" }, "Contrast"),
                    react_1.default.createElement(Button_1.default, { color: "contrast", disabled: true }, "Contrast")),
                react_1.default.createElement(muix_primitives_1.View, { classes: { root: rootView } },
                    react_1.default.createElement(Button_1.default, { raised: true }, "Default"),
                    react_1.default.createElement(Button_1.default, { raised: true, color: "primary" }, "Primary"),
                    react_1.default.createElement(Button_1.default, { raised: true, color: "accent" }, "Accent"),
                    react_1.default.createElement(Button_1.default, { raised: true, color: "contrast" }, "Contrast"),
                    react_1.default.createElement(Button_1.default, { raised: true, color: "accent", disabled: true }, "Disabled"),
                    react_1.default.createElement(Button_1.default, { raised: true, dense: true }, "Dense")),
                react_1.default.createElement(muix_primitives_1.View, { classes: { root: rootView } },
                    react_1.default.createElement(Button_1.default, { fab: true, color: "primary" },
                        react_1.default.createElement(muix_primitives_1.Icon, null, "M19,13H13V19H11V13H5V11H11V5H13V11H19V13Z" /* Plus */)),
                    react_1.default.createElement(Button_1.default, { fab: true, mini: true, color: "primary" },
                        react_1.default.createElement(muix_primitives_1.Icon, null, "M19,13H13V19H11V13H5V11H11V5H13V11H19V13Z" /* Plus */)),
                    react_1.default.createElement(Button_1.default, { fab: true, color: "accent" },
                        react_1.default.createElement(muix_primitives_1.Icon, null, "M20.71,7.04C21.1,6.65 21.1,6 20.71,5.63L18.37,3.29C18,2.9 17.35,2.9 16.96,3.29L15.12,5.12L18.87,8.87M3,17.25V21H6.75L17.81,9.93L14.06,6.18L3,17.25Z" /* Pencil */)),
                    react_1.default.createElement(Button_1.default, { fab: true, mini: true, color: "accent" },
                        react_1.default.createElement(muix_primitives_1.Icon, null, "M20.71,7.04C21.1,6.65 21.1,6 20.71,5.63L18.37,3.29C18,2.9 17.35,2.9 16.96,3.29L15.12,5.12L18.87,8.87M3,17.25V21H6.75L17.81,9.93L14.06,6.18L3,17.25Z" /* Pencil */)),
                    react_1.default.createElement(Button_1.default, { fab: true, disabled: true },
                        react_1.default.createElement(muix_primitives_1.Icon, null, "M19,4H15.5L14.5,3H9.5L8.5,4H5V6H19M6,19C6,20.1 6.9,21 8,21H16C17.1,21 18,20.1 18,19V7H6V19Z" /* Delete */)),
                    react_1.default.createElement(Button_1.default, { fab: true, mini: true, disabled: true },
                        react_1.default.createElement(muix_primitives_1.Icon, null, "M19,4H15.5L14.5,3H9.5L8.5,4H5V6H19M6,19C6,20.1 6.9,21 8,21H16C17.1,21 18,20.1 18,19V7H6V19Z" /* Delete */))),
                react_1.default.createElement(muix_primitives_1.View, { classes: { root: rootView } },
                    react_1.default.createElement(Button_1.ButtonIconLeft, { color: 'accent', raised: true },
                        react_1.default.createElement(muix_primitives_1.Icon, { children: "M5,3H19C20.1,3 21,3.9 21,5V19C21,20.1 20.1,21 19,21H5C3.9,21 3,20.1 3,19V5C3,3.9 3.9,3 5,3M12,17L17,12H14V8H10V12H7L12,17Z" /* ArrowDownBoldBox */ }),
                        "Icon Left"),
                    react_1.default.createElement(Button_1.ButtonIconLeft, { color: 'primary', raised: true },
                        react_1.default.createElement(muix_primitives_1.Icon, { children: "M5,3H19C20.1,3 21,3.9 21,5V19C21,20.1 20.1,21 19,21H5C3.9,21 3,20.1 3,19V5C3,3.9 3.9,3 5,3M12,17L17,12H14V8H10V12H7L12,17Z" /* ArrowDownBoldBox */ }),
                        "Icon Left"),
                    react_1.default.createElement(Button_1.ButtonIconRight, { color: 'default', raised: true },
                        "Icon Right",
                        react_1.default.createElement(muix_primitives_1.Icon, { children: "M5,3H19C20.1,3 21,3.9 21,5V19C21,20.1 20.1,21 19,21H5C3.9,21 3,20.1 3,19V5C3,3.9 3.9,3 5,3M12,17L17,12H14V8H10V12H7L12,17Z" /* ArrowDownBoldBox */ })),
                    react_1.default.createElement(Button_1.ButtonIconRight, { color: 'contrast', dense: true, raised: true },
                        "Icon Right",
                        react_1.default.createElement(muix_primitives_1.Icon, { children: "M5,3H19C20.1,3 21,3.9 21,5V19C21,20.1 20.1,21 19,21H5C3.9,21 3,20.1 3,19V5C3,3.9 3.9,3 5,3M12,17L17,12H14V8H10V12H7L12,17Z" /* ArrowDownBoldBox */ })),
                    react_1.default.createElement(Button_1.ButtonIconRight, { color: 'primary', disabled: true, raised: true },
                        "Icon Right",
                        react_1.default.createElement(muix_primitives_1.Icon, { children: "M5,3H19C20.1,3 21,3.9 21,5V19C21,20.1 20.1,21 19,21H5C3.9,21 3,20.1 3,19V5C3,3.9 3.9,3 5,3M12,17L17,12H14V8H10V12H7L12,17Z" /* ArrowDownBoldBox */ }))),
                react_1.default.createElement(muix_primitives_1.View, { classes: { root: rootView } },
                    react_1.default.createElement(Button_1.default, { color: 'accent', raised: true, classes: function (theme) { return ({ raisedAccent: { backgroundColor: 'green', $web: { '&:hover': { backgroundColor: 'green', } } } }); } }, "classes"),
                    react_1.default.createElement(muix_styles_1.MuiThemeProvider, { theme: theme },
                        react_1.default.createElement(Button_1.default, { color: 'accent', raised: true }, "theme")))); };
            exports_1("default", app);
        }
    };
});
