// muix/common/shadows/native.js
//for testing with https://snack.expo.io/
System.register(["tslib", "react", "react-native"], function (exports_1, context_1) {
    "use strict";
    var __moduleName = context_1 && context_1.id;
    var tslib_1, react_1, react_native_1, android, iosW, iosH, iosOpacity, iosRadius, getNativeShadow, style, AppComp;
    return {
        setters: [
            function (tslib_1_1) {
                tslib_1 = tslib_1_1;
            },
            function (react_1_1) {
                react_1 = react_1_1;
            },
            function (react_native_1_1) {
                react_native_1 = react_native_1_1;
            }
        ],
        execute: function () {
            android = [0, 2, 4, 9, 12, 15, 18, 20, 22, 24, 26, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40];
            iosW = function (deep) {
                switch (deep) {
                    case 0:
                    case 1: return 0;
                    default: return deep / 10;
                }
            };
            iosH = function (deep) {
                switch (deep) {
                    case 0: return 0;
                    default: return 0.7 + (deep - 1) * 0.7;
                }
            };
            iosOpacity = function (deep) {
                switch (deep) {
                    case 0:
                    case 1: return 0.7;
                    default: return 0.7 - deep * 0.02;
                }
            };
            iosRadius = function (deep) {
                switch (deep) {
                    case 0: return 2;
                    case 1: return 3;
                    default: return 4 + deep * 0.1;
                }
            };
            getNativeShadow = function (deep) {
                if (deep == 0)
                    return {};
                return {
                    shadowColor: 'black',
                    shadowOffset: { width: iosW(deep), height: iosH(deep) },
                    shadowOpacity: iosOpacity(deep),
                    shadowRadius: iosRadius(deep),
                    elevation: android[deep],
                };
            };
            style = {
                width: 50,
                height: 50,
                margin: 20,
                borderWidth: 1,
                borderColor: 'brown',
                borderStyle: 'solid',
                backgroundColor: 'black',
                flexShrink: 0,
            };
            AppComp = /** @class */ (function (_super) {
                tslib_1.__extends(AppComp, _super);
                function AppComp() {
                    return _super !== null && _super.apply(this, arguments) || this;
                }
                AppComp.prototype.render = function () {
                    return react_1.default.createElement(react_native_1.ScrollView, { style: { paddingTop: 20 } },
                        react_1.default.createElement(react_native_1.View, { style: { flexDirection: 'row', flexWrap: 'wrap' } }, android.map(function (sh, idx) { return react_1.default.createElement(react_native_1.View, { key: idx, style: tslib_1.__assign({}, style, getNativeShadow(idx)) }); })));
                };
                return AppComp;
            }(react_1.default.Component));
            exports_1("default", AppComp);
        }
    };
});
